source/AI.cpp

```diff
@@ -1710,7 +1710,28 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, const System *playerSy
 	if(ship.Health() < minHealth && (!ship.IsYours() || Preferences::Has("Damaged fighters retreat")))
 		return true;
 
-	// TODO: Reboard if in need of ammo.
+	// If a fighter is armed with only ammo-using weapons, but no longer has the ammunition
+	// needed to use them, it should dock if the parent can supply that ammo.
```

Comment:
Should this be preference based? Or should we just change it so that any ammo-using weapon that's out of ammo is enough for a fighter to restock? Although a fighter that has non-ammo-using weapons combined with ammo-using weapons doesn't need to restock in order to continue fighting, such a fighter with ammo for its ammo-using weapons will be more effective than one without, so it can be beneficial to restock even if it isn't necessary to continue fighting.

Reply:
I don't know that it has a meaningful impact, and it is certainly something that can be added later if there is demand for it.

Reply:
I'm fine with considering this for later.